contents/javascript/Javascript-framework-Angular.md

```diff
@@ -0,0 +1,231 @@
+<frontmatter>
+  title: Introduction to Angular
+  header: pagetop.md
+  footer: footer.md
+  head: head.md
+  siteNav: mainNav.md
+  pageNav: 3
+</frontmatter>
+
+<div class="website-content">
+
+{{ booktitle | safe }}
+
+# Introduction to Angular
+
+**Author: [Ronak Lakhotia](https://github.com/RonakLakhotia)**                                                              
+Reviewers: [Tan Heng Yeow](https://github.com/tanhengyeow), [Jacob Li PengCheng](https://github.com/jacoblipech)
+
+## What is Angular?
+
+The [official website](https://angular.io/docs) describes Angular as follows:
+
+>Angular is a platform that makes it easy to build applications with the web. Angular combines declarative templates, dependency injection, end to end tooling, and integrated best practices to solve development challenges. Angular empowers developers to build applications that live on the web, mobile, or the desktop.
+
+Let us understand some key terms in that definition starting with _declarative templates_.
+Consider a simple web application that displays the name of a person as shown below.
+
+<iframe src="https://stackblitz.com/edit/angular-3qy4j2?embed=1&file=src/app/app.component.ts&view=preview" style="width:85%; height:380px; border:0; border-radius: 4px; overflow:hidden;" sandbox="allow-modals allow-forms allow-popups allow-scripts allow-same-origin"></iframe> 
+
+<box type="info">
+    The sandbox allows you to edit the code of the web application. You can enter your name in the text box and see how the UI is automatically updated. You can also view and edit the code by clicking on the "Both" button. 
+</box>
+
+### Templates and Components
+
+The UI of an Angular app is defined in _templates_ which contain a mix of HTML code and HTML-like [custom tags](https://angular.io/guide/elements). The template for the above example (defined in `person.component.html`) is given below:
+
+``` html
+<h1>About Me!</h1>
+<h2>Hello World! My name is {person.name}</h2>
+<div>
+<label>NAME: <input [(ngModel)]="person.name"/></label>
+</div>
+```
+  
+The templates used in Angular are _declarative_ because they describe _what_ the view should look like rather than give instructions on _how_ to construct the view. In the above example, we declare what we want to display (i.e., name) and Angular renders the view. A noteworthy feature about the UI rendered by Angular is its _dynamic behavior_.
+You will notice that as you change the name in the text box, the name rendered on the app is also updated. This is because the UI and the application data are automatically synchronized by Angular as explained [here](https://angular.io/guide/template-syntax#binding-syntax-an-overview).
+
+Angular organizes the <tooltip content="Code that describes _how_ the view must rendered">behavior instructions</tooltip> (written in TypeScript) into what it calls _Components_. Given below is the component file `person.component.ts` of the `PersonComponent` :  
```

Comment:
I can't seem to find a related jargon named "behavior instructions" so it feels like a made up term. It might be better to just replace this term with what you explained in the tooltip.